### PR TITLE
Add integration tests for mSupply Query and Web UI queries

### LINKED ISSUE #423: Setup integration tests for existing API
Particularly MSupply Queries

Things to note!

mSupply uses port 2048
https://codes.msupply.foundation:2048/v1/graphql/

It also mentions another server
https://api.universalcodes.msupply.org.nz
This appears to be a different IP Address...

V1 API here???
https://api.universalcodes.msupply.org.nz/v1/items?search=ibup&exact=false

### PR DESCRIPTION
**Pull request by jmbrunskill**
Fixes #423 

## Description

Adds Integration Tests for data from the current UI, as well as mSupply Query format.

Not sure how deeply to test. The main thing is that right codes are returned (is my assumption)
Would be nice to have the tests pass if extra data is returned (e.g. new codes are added) not sure how reliable that is though?

Would be nice to put this test behind some kind of integration flag or group or somethings. I assume we can tackle that as part of removing `nx` though.

## Checklist:

- [x] Deleted e2e test which uses cypress? and was never setup
- [x] Adjusted jest.conf.ts files to make them vaguely work
- [ ] Haven't fixed the `npx nx test` issue!

### Documentation:

- [ ] Run using `npx jest`

### CODE REVIEW

File: unified-codes/apps/data-service/src/app/integration/v2_graphql.test.ts [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+            code: 'c8ba31a5',
+            description: 'Acetylcysteine',
+            type: 'drug',
+            uid: '0x186da',
+          },
+          {
+            code: 'cf00cc3f',
+            description: 'Acetylsalicylic Acid',
+            type: 'drug',
+            uid: '0x186e6',
```
Comment (lache-melvin) [inline unified-codes/apps/data-service/src/app/integration/v2_graphql.test.ts:187]: These uids were different for me, when I compared `res` and `expected`... can use `expect.any(String)` if you want to test the field exists though 👍

**Review by lache-melvin — APPROVED** [review]
Noticed this PR is going into `master`, rather than `develop`, which I keep doing too haha - might be worth changing the default branch if you have access to the settings for this repo?

Added a few comments but will be awesome to have this in there! 🚀

**Comment by jmbrunskill** [comment]
> might be worth changing the default branch if you have access to the settings for this repo?

Done!

File: unified-codes/apps/data-service/src/app/integration/v2_graphql.test.ts [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    .then((res) => res.json())
+    .then((res) => {
+      expect(res.data.entity.code).toEqual('c7750265');
+      expect(res.data.entity.description).toEqual('Abacavir');
+      expect(JSON.stringify(res)).toContain('888e7a00');
+      expect(JSON.stringify(res)).toContain('db3d2000');
+      expect(JSON.stringify(res)).toContain('358b04bf');
+      expect(JSON.stringify(res)).toContain('945b5500');
+      expect(JSON.stringify(res)).toContain('3590a4bf');
+      expect(res.data.entity.properties).toEqual(expected.data.entity.properties);
```
[the remark this answers is not in the document]
Comment (jmbrunskill) [reply unified-codes/apps/data-service/src/app/integration/v2_graphql.test.ts:156]: I was worried about things being added to the drug, but I think you're right.
I've added a comment as well.

File: unified-codes/apps/data-service/src/app/integration/v2_graphql.test.ts (hunk 1 above)
Comment (jmbrunskill) [reply unified-codes/apps/data-service/src/app/integration/v2_graphql.test.ts:187]: I don't think we need to worry about uid, interesting that it's different every time. Makes it kind of useless to return...

File: unified-codes/apps/data-service/src/app/integration/v2_graphql.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jmbrunskill) [reply unified-codes/apps/data-service/src/app/integration/v2_graphql.test.ts:162]: [43e92a1](https://github.com/msupply-foundation/unified-codes/pull/425/commits/43e92a197d896d44ef347a914449e1acd5c871ef)

**Pull request merged**